src/components/app/index.tsx

```diff
@@ -93,6 +98,19 @@ export default class App extends Component<Props, State> {
     const fileInput = event.target as HTMLInputElement;
     const sourceFile = fileInput.files && fileInput.files[0];
     if (!sourceFile) return;
+    await this.updateFile(sourceFile);
+  }
+
+  @bind
+  async onFileDrop(event: CustomEvent) {
+    const sourceFile = event.detail as File;
+    this.setState(initialState());
```

Comment:
I wouldn't expect dropping a file to change my code settings.

Reply:
So.... there were some issues. This reset's everything back to what the page was on first load, there's a heap of state wrapped up with images, codecs, and other changes that I didn't feel like they applied when I dropped a new image on to the canvas.